pkg/sshd/lock.go

```diff
@@ -0,0 +1,68 @@
+package sshd
+
+import (
+	"fmt"
+	"sync"
+)
+
+// RepositoryLock interface that allows the creation of a lock associated
+// with a repository name to avoid simultaneous git operations
+type RepositoryLock interface {
+	// Lock aquires a lock for a repository
+	Lock(repoName string) error
```

Comment:
@arschles I tried that but it feels awkward see a git push just waiting. The behavior in this PR provides the same UX than in deis V1